docs/src/lib/functions.md

```diff
@@ -4,6 +4,34 @@ CurrentModule = DataFrames
 
 # Functions
 
+## Multi-threading support
+
+Selected operations in DataFrames.jl support muliti-threading.
+It is task-based and implemented using the `@spawn` macro form Julia Base.
+
+This is a list of operations that currently support multi-threading:
+- `DataFrame` constructor with `copycols=true`; also recursively all functions
+  that call this constructor, e.g. `copy`.
+- `getindex` when multiple columns are selected.
+- `groupby` (both when hashing is required and when fast path using `DataAPI.refpool`
+  is used).
+- `join*` functions for composing output data frame (but currently not for finding
+  matching rows in joined data frames).
+- in `combine`, `select[!]``, and `transform[!]` on `GroupedDataFrame` when performing
+  multiple transformations (each transformation is spawned in a separate task)
+- in `combine` when a transformation produces one row per group and the passed tranformation
```

Comment:
Any comment on this?

Reply:
Ah - I missed the question. I was thinking if we should include this. This can happen only if `select` or `transform` are passed a `GroupedDataFrame` that has exactly one row per group and when aggregation also produces one row per group. This is an unlikely scenario (in this scenario it does make sense to do `groupby` and then `select`/`transform` but rather one should just run `select`/`transform` on a data frame with `ByRow`). Do you think we should add `select` and `transform` to the list (as technically they could, but the condition when they would is unlikely).

In essence since this is "one row per group" case it means that we are `combine`ing rows unless each group has exactly one row to start with 😄.

Reply:
@nalimilan - what do you think given this comment?

Reply:
Sorry, I had missed your reply. Isn't multithreading also used with `select` and `transform` when it returns a single row for each group (disregarding the number of rows in the group)? We repeat the returned row to match the number of rows in the input, but the step before that is multithreaded IIRC, right?

Reply:
You are right (as usual - I am glad to have you :smile:). I have fixed this.